conftest.py

```diff
@@ -79,12 +80,21 @@ def small_cluster(request):
 
 
 @pytest.fixture
-def small_client(small_cluster):
+def small_client(small_cluster, s3, s3_scratch, s3_cluster_dump_url):
     with Client(small_cluster) as client:
         small_cluster.scale(10)
         client.wait_for_workers(10)
         client.restart()
-        yield client
+        try:
+            yield client
+        except Exception:
+            cluster_dump = strtobool(os.environ.get("CLUSTER_DUMP", "false"))
+            print(f"{cluster_dump=}")
+            if cluster_dump:
+                print(f"Cluster state dump can be found at: {s3_cluster_dump_url}")
+                client.dump_cluster_state(s3_cluster_dump_url)
+            else:
+                pass
```

Comment:
I think we'll still want to `raise` the exception here so that test failures show up 

```suggestion
            raise
```